File: 2 pointers/1535_find_the_winner_of_an_array_game.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var getWinner = function(arr, k) {

    let l = 0, r = 1
    let curr = 0, count = 0

    while(r < arr.length){

        if(arr[l] > arr[r]){
            count ++
            r ++
        } else {
            curr = r
            l = r
            r = l+1
            count = 1
        }
        // console.log(l,r,curr,count)
        if(count >= k) return arr[curr]
    }

    return arr[curr]
};

File: 2 pointers/1535_find_the_winner_of_an_array_game.go
func getWinner(arr []int, k int) int {

    l := 0
    r := 1
    curr := 0
    count := 0

    for r < len(arr) {

        if arr[l] > arr[r] {
            r ++
            count ++
        } else {
            l = r
            curr = r
            r = r+1
            count = 1
        }

        if count >= k {
            return arr[curr]
        }
    }

    return arr[curr]
}